fe/fe-core/src/main/java/com/starrocks/memory/MemoryTrackable.java

```diff
@@ -14,18 +14,28 @@
 
 package com.starrocks.memory;
 
+import com.starrocks.common.Pair;
 import org.apache.spark.util.SizeEstimator;
 
-import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public interface MemoryTrackable {
-
     default long estimateSize() {
-        return SizeEstimator.estimate(this);
-    }
+        List<Pair<List<Object>, Long>> samples = getSamples();
+        long totalBytes = 0;
+        for (Pair<List<Object>, Long> pair : samples) {
+            List<Object> sample = pair.first;
+            long size = pair.second;
+            if (sample.size() > 0) {
```

Comment:
Precondition.check?

Reply:
No, empty is normal when there is no objects in that manager.